List Basic information on available topologies

aarondees-blabs opened the pull request:
Added functionality to return a list of basic information on all available topologies.

Change Log:
 - ListTopologiesCreateRequest struct added
 - BasicTopologyInfo struct added
 - GetAllTopologies() function added, returns an array of
 BasicTopologyInfo

Commit:
Added functionality to return a list of basic information on all
available topologies.

Change Log:
 - ListTopologiesCreateRequest struct added
 - BasicTopologyInfo struct added
 - GetAllTopologies() function added, returns an array of
 BasicTopologyInfo

laurentganne commented on alien4cloud/alien4cloud_structs.go:
```diff
@@ -329,6 +329,12 @@ type DeploymentArtifact struct {
 	Description          string                 `json:"description,omitempty"`
 }
 
+type ListTopologiesCreateRequest struct {
```
This new type may not be needed as the type searchRequest described above is identical

aarondees-blabs replied:
Yes, I agree. Done.

laurentganne commented on alien4cloud/topology.go:
```diff
@@ -517,3 +518,55 @@ func (t *topologyService) SaveA4CTopology(ctx context.Context, a4cCtx *TopologyE
+		[]byte(string(getTopoJSON)),
+		[]Header{
+			{
+				"Content-Type",
```
Loïc introduced recently constants for headers, you could use:
```
[]Header{contentTypeAppJSONHeader, acceptAppJSONHeader},
```

aarondees-blabs replied:
Done.

laurentganne commented on alien4cloud/topology.go:
```diff
@@ -517,3 +518,55 @@ func (t *topologyService) SaveA4CTopology(ctx context.Context, a4cCtx *TopologyE
 	return processA4CResponse(response, nil, http.StatusOK)
 }
+
+func (t *topologyService) GetAllTopologies(ctx context.Context) ([]BasicTopologyInfo, error) {
```
To have a more generic API similar to the REST API, we could name this function `GetTopologies()` and add the argument `query string` that you would pass below in the searchRequest.query parameter.

aarondees-blabs replied:
Sounds good, done.

Commit:
Changes after code review.

Change log:
 - Added constants for headers
 - Removed ListTopologiesCreateRequest, used searchRequest instead.
 - Reamed GetAllTopologies to GetTopologies, with argument query which
 is passed to searchRequest.

Commit:
Updated, now with changes to topology.go

laurentganne committed:
Added unit test

laurentganne reviewed (approved):
Looks good!
